plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/actions/DataSourcePropertyTester.java

```diff
@@ -61,9 +65,14 @@ public boolean test(Object receiver, String property, Object[] args, Object expe
             if (!(receiver instanceof DBPContextProvider)) {
                 return false;
             }
-            DBPContextProvider contextProvider = (DBPContextProvider)receiver;
+            DBPContextProvider contextProvider = (DBPContextProvider) receiver;
             @Nullable
             DBCExecutionContext context = contextProvider.getExecutionContext();
+            @Nullable
+            DBPProject resourceProject = receiver instanceof IEditorPart
+                ? EditorUtils.getFileProject(((IEditorPart)receiver).getEditorInput())

```

Comment:
The `EditorUtils#getFileProject` method can fail and return `null`. It would be better if there was a nullability check.

```suggestion
                ? EditorUtils.getFileProject(((IEditorPart) receiver).getEditorInput())

```